pkg/controller/deactivation/deactivation_controller_test.go

```diff
@@ -0,0 +1,421 @@
+			// then
+			require.Error(t, err)
+			require.Contains(t, err.Error(), "usersignup get error")
+			require.False(t, res.Requeue, "requeue should not be set")
+			require.True(t, res.RequeueAfter == 0, "requeueAfter should not be set")
+		})
+
+		// cannot update UserSignup
+		t.Run("UserSignup update failure", func(t *testing.T) {
+			// given
+			expectedDeactivationTimeout := 30
+			userSignup := &toolchainv1alpha1.UserSignup{
+				ObjectMeta: newObjectMeta(username, "foo@bar.com"),
+				Spec: toolchainv1alpha1.UserSignupSpec{
+					Username:      "foo@bar.com",
+					Approved:      true,
+					TargetCluster: "east",
+				},
+			}
+			basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+			murProvisionedTime := &metav1.Time{Time: time.Now().Add(-time.Duration(expectedDeactivationTimeout*24) * time.Hour)}
+			mur := murtest.NewMasterUserRecord(t, username, murtest.Account("cluster1", *basicTier), murtest.ProvisionedMur(murProvisionedTime), murtest.UserIDFromUserSignup(userSignup))
+			initObjs := []runtime.Object{basicTier, mur, userSignup}
+			r, req, cl := prepareReconcile(t, mur.Name, initObjs...)
+			cl.MockUpdate = func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
+				_, ok := obj.(*toolchainv1alpha1.UserSignup)
+				if ok {
+					return fmt.Errorf("usersignup update error")
+				}
+				return nil
+			}
+			// when
+			res, err := r.Reconcile(req)
+			// then
+			require.Error(t, err)
+			require.Contains(t, err.Error(), "usersignup update error")
+			require.False(t, res.Requeue, "requeue should not be set")
+			require.True(t, res.RequeueAfter == 0, "requeueAfter should not be set")
+			deactivated, err := isUserSignupDeactivated(cl, username)
+			require.NoError(t, err)
+			require.False(t, deactivated)
+		})
+	})
+
+}
+
+func prepareReconcile(t *testing.T, name string, initObjs ...runtime.Object) (reconcile.Reconciler, reconcile.Request, *test.FakeClient) {
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	cl := test.NewFakeClient(t, initObjs...)
+	cfg, err := configuration.LoadConfig(cl)
+	r := &ReconcileDeactivation{client: cl, scheme: s, config: cfg}
+	return r, reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: operatorNamespace,
+		},
+	}, cl
+}
+
+func newObjectMeta(name, email string) metav1.ObjectMeta {
+	if name == "" {
+		name = uuid.NewV4().String()
+	}
+
+	md5hash := md5.New()
+	// Ignore the error, as this implementation cannot return one
+	_, _ = md5hash.Write([]byte(email))
+	emailHash := hex.EncodeToString(md5hash.Sum(nil))
+
+	return metav1.ObjectMeta{
+		Name:      name,
+		Namespace: operatorNamespace,
+		Annotations: map[string]string{
+			toolchainv1alpha1.UserSignupUserEmailAnnotationKey: email,
+		},
+		Labels: map[string]string{
+			toolchainv1alpha1.UserSignupUserEmailHashLabelKey: emailHash,
+		},
+	}
+}
+
+func isUserSignupDeactivated(cl *test.FakeClient, name string) (bool, error) {
```

Comment:
we could make it an assertion method 
`func assertThatUserSignupDeactivated(t *testing.T, cl *test.FakeClient, name string, expectedValue bool)`
it would make the tests shorter

Reply:
Also a good point, done in 990b83d5fa5e4071eb93d641b014e07a6d006004